carbon-design-system/carbon-addons-iot-react
# 1181 piechart dashboard card

Closes #1181 

**Summary**
Added a new PieChartCard
![image](https://user-images.githubusercontent.com/5167091/94196362-852b5780-feb4-11ea-97c4-de755a5ea566.png)

**Change List (commits, features, bugs, etc)**
- Added a new card called PieChartCard that contains a PieChart and a Table
- Using Table instead of StatefulTable like the other chart cards since there is no need for a state for this simple table
- There is no series prop, just a values prop that contains slice objects with both the label and the value since this is what the underlying PieChart api is expecting. The actual value must be called "value" but the label prop can be called anything and be controlled via 'groupDataSourceId'
- Added the properties for the expected customisations but also added the overrides pattern since due to the lack of actual specs
- Added a small addition to the overrides pattern so that a prop now also can be a function that returns an object, instead of just being an object. Added a small helper function in componentUtilityFunctions.js to check this.
- Created a card utility function called increaseSmallCardSize that is shared between the BarChartCard and this new new PieChartCard
- Added unit tests

**Acceptance Test (how to verify the PR)**
1. Go to '?path=/story/watson-iot-piechartcard--custom-tooltip' 
2. Verify that the different stories work and make sense
